platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowGraphSynchronizer.java

```diff
@@ -347,48 +384,71 @@ private List<EventImpl> createSendList(
 
         return sendList;
     }
 
     /**
-     * Executes phase 3 of a sync
+     * By this point in time, we have figured out which events we want to send the peer, and the peer has figured out
+     * which events it wants to send us. In parallel, send and receive those events.
      *
-     * @param conn     the connection to use
-     * @param timing   metrics that track sync timing
-     * @param sendList the events to send
+     * @param connection          the connection to use
+     * @param timing              metrics that track sync timing
+     * @param sendList            the events to send
+     * @param syncKeepAlivePeriod the period at which the reading thread should send keepalive messages
+     * @param maxSyncTime         the maximum amount of time to spend syncing with a peer, syncs that take longer than
+     *                            this will be aborted
      * @return true if the phase was successful, false if it was aborted
      * @throws ParallelExecutionException if anything goes wrong
      */
-    private boolean phase3(final Connection conn, final SyncTiming timing, final List<EventImpl> sendList)
+    private boolean sendAndReceiveEvents(
+            @NonNull final Connection connection,
+            @NonNull final SyncTiming timing,
+            @NonNull final List<EventImpl> sendList,
+            @NonNull final Duration syncKeepAlivePeriod,
+            @NonNull final Duration maxSyncTime)
             throws ParallelExecutionException {
+
+        Objects.requireNonNull(connection);
+        Objects.requireNonNull(sendList);
+
         timing.setTimePoint(4);
         // the reading thread uses this to indicate to the writing thread that it is done
         final CountDownLatch eventReadingDone = new CountDownLatch(1);
         // the writer will set it to true if writing is aborted
         final AtomicBoolean writeAborted = new AtomicBoolean(false);
         final Integer eventsRead = readWriteParallel(
-                SyncComms.phase3Read(conn, eventHandler, syncMetrics, eventReadingDone, intakeEventCounter),
-                SyncComms.phase3Write(conn, sendList, eventReadingDone, writeAborted),
-                conn);
+                readEventsINeed(
+                        connection, eventHandler, syncMetrics, eventReadingDone, intakeEventCounter, maxSyncTime),
+                sendEventsTheyNeed(connection, sendList, eventReadingDone, writeAborted, syncKeepAlivePeriod),
+                connection);
         if (eventsRead < 0 || writeAborted.get()) {
             // sync was aborted
-            logger.info(SYNC_INFO.getMarker(), "{} sync aborted", conn::getDescription);
             return false;
         }
-        logger.info(
-                SYNC_INFO.getMarker(), "{} writing events done, wrote {} events", conn::getDescription, sendList::size);
-        logger.info(SYNC_INFO.getMarker(), "{} reading events done, read {} events", conn.getDescription(), eventsRead);
```

Comment:
I don't think its a good idea to remove sync logs

Reply:
I 100% agree.  Removing logs should only happen when there is no one left who needs to see the logs as part of root cause analysis when something goes wrong.

Reply:
added back in